compliance/collection/nodeinventorizer/nodeinventory.go

```diff
@@ -63,10 +62,25 @@ func convertRHELComponents(rc *database.RHELv2Components) []*scannerV1.RHELCompo
 		log.Warn("No RHEL packages found in scan result")
 		return nil
 	}
+
+	seenEntries := make(map[int]*database.RHELv2Package)
 	v1rhelc := make([]*scannerV1.RHELComponent, 0, len(rc.Packages))
-	for _, rhelc := range rc.Packages {
+
+	// referencing this label by the inner `continue` lets us skip the rest of the outer loop as well,
+	// resulting in the colliding component not being added to v1rhelc.
+COMPONENTS:
+	for i, rhelc := range rc.Packages {
+		for _, candidate := range seenEntries {
+			if equalRHELv2Packages(candidate, rhelc) {
+				log.Warnf("Detected package collision in Node Inventory scan. Skipping package %v for id %v", candidate, i)
+				continue COMPONENTS
+			}
+		}
+		// If the entry didn't produce a collision, add it to seenEntries and result
+		seenEntries[i] = rhelc
+		log.Debugf("Adding component %v to v1rhelc", rhelc.Name)
 		v1rhelc = append(v1rhelc, &scannerV1.RHELComponent{
-			// TODO(ROX-13936): Find out if ID field is needed here
+			Id:          int64(i),
```

Comment:
A quick comment on why the index is being used as the ID would be helpful.

Reply:
Done!